### PR TITLE
Marko/tm cmn to db

### PR DESCRIPTION
**Pull request by marbar3778**

### CODE REVIEW

File: db/remotedb/grpcdb/client.go [hunk 1]
```diff
@@ -4,7 +4,7 @@ import (
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials"
 
-	protodb "github.com/tendermint/tm-cmn/db/remotedb/proto"
+	protodb "github.com/tendermint/tm-db/db/remotedb/proto"
```
Comment (melekes) [inline db/remotedb/grpcdb/client.go]: Can we get rid of `db` directory and move its content into the root dir?

File: util.go [hunk 2]
```diff
@@ -43,3 +45,36 @@ func IsKeyInDomain(key, start, end []byte) bool {
...
+	_, err := os.Stat(filePath)
+	return !os.IsNotExist(err)
+}
+
+const (
+	strChars = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz" // 62 characters
+)
+
+// Str constructs a random alphanumeric string of given length.
+func RandStr(length int) string {
```
Comment (melekes) [inline util.go]: Do we use this in tests? If yes, then we should unexport it and move into `test_helpers.go`

File: util.go [hunk 3]
```diff
@@ -43,3 +45,36 @@ func IsKeyInDomain(key, start, end []byte) bool {
 	}
 	return true
 }
+
+func FileExists(filePath string) bool {
```
Comment (melekes) [inline util.go]: same here. should probably be unexported

Comment (marbar3778) [reply util.go]: used in fsdb.go

File: util.go (hunk 2 above)
Comment (marbar3778) [reply util.go]: will move this to test_helpers.go

**Pull request merged**